docs/development/go/go-structures/index.md

```diff
@@ -18,235 +19,532 @@ external_resources:
 
 ## Introduction
 
-Although arrays, slices and maps are all very useful, they cannot group and hold values of multiple data types at the same place. When you need to group various types of variables and create new data types, you can use *structures* – the various elements of a structure are called *fields* of the structure or just fields. For reasons that will become clear in a while, the fields of a structure usually begin with an uppercase letter. Last, the order in which you put the fields in the definition of a structure type is significant for the type identity of the defined structure.
+Go's array, slice, and map types can be used to group multiple elements, but they cannot hold values of multiple data types. When you need to group different types of variables and create new data types, you can use *structs*.
 
 {{< note >}}
-This guide is written for a non-root user – none of the presented commands requires elevated privileges.
+Go does not have a concept of *classes* from other object oriented languages. Structs will be used in similar ways as classes, with important differences. For example, there is no class inheritance feature in Go.
 {{< /note >}}
 
-## In This Guide
+In this guide you will:
 
-This guide you teach you about creating and using structures in Go, the `new` keyword, pointers to structures and implementing functions that work with structures. All you need to follow this guide is to have Go and your favorite text editor installed on your Linode machine.
+- [Review a simple introductory struct](#a-simple-struct) and learn about basic struct semantics.
+- Find out how to use [pointers to structs](#pointers-and-structs).
+- Implement [methods with structs](#methods).
+- Explore the different ways to [instantiate structs](#creating-structs).
+- Read through an example of how to [encode and decode JSON](#structs-and-json) with structs.
 
-## A simple structure
+## Before You Begin
 
-The following Go program defines and uses a new structure type called `XYZ`.
+To run the examples in this guide, your workstation or server will need to have Go installed, and the `go` CLI will need to be set in your terminal's PATH:
 
-{{< file "./structures.go" go >}}
+- If you use Ubuntu, follow our [How to Install Go on Ubuntu](/docs/development/go/install-go-on-ubuntu/) guide.
+- Follow the [Getting Started](https://golang.org/doc/install) guide on Golang's website to install on other operating systems.
+
+An introductory-level knowledge of Go is assumed by this guide. If you're just getting started with Go, check out our [Learning Go Functions, Loops, and Errors](/docs/development/go/learning-go-functions-loops-and-errors-a-tutorial/) tutorial.
+
+## A Simple Struct
```

Comment:
The top-line changes to the A Simple Struct section are:

- Re-named the example struct and fields so that they more clearly reflect the naming conventions in Go: camel-case, importance of first-letter capitalization. I thought it was a bit harder to see these conventions with one-letter field names.
- Split the single example into several smaller examples that illustrate different parts of the example code. I did this split because we generally prefer to isolate new concepts as they are introduced:
    - The different syntaxes for creating a struct from a struct literal
    - How to use dot notation
    - Field name capitalization and public/private fields
    - Copy/value semantics